### ISSUE TITLE
Private packages fail to install on 0.28.1

### ISSUE THREAD

**Issue by AlastairTaft** [yarnpkg/yarn#3998]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**
When I deploy on heroku, it automatically uses the latest version 0.28.1 and the deploy fails due to not being able to find private packages. 

```cms
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Node.js app detected
remote: 
remote: -----> Creating runtime environment
remote:        
remote:        NPM_CONFIG_LOGLEVEL=error
remote:        NPM_CONFIG_PRODUCTION=true
remote:        NODE_VERBOSE=false
remote:        NODE_ENV=production
remote:        NODE_MODULES_CACHE=false
remote: 
remote: -----> Installing binaries
remote:        engines.node (package.json):  unspecified
remote:        engines.npm (package.json):   4
remote:        engines.yarn (package.json):  unspecified (use default)
remote:        
remote:        Resolving node version 6.x via semver.io...
remote:        Downloading and installing node 6.11.1...
remote:        Resolving npm version 4 via semver.io...
remote:        Downloading and installing npm 4.6.1 (replacing version 3.10.10)...
remote:        Resolving yarn version (latest)...
remote:        Downloading and installing yarn (0.28.1)...
remote:        Installed yarn 0.28.1
remote: 
remote: -----> Restoring cache
remote:        Skipping cache restore (disabled)
remote: 
remote: -----> Building dependencies
remote:        Installing node modules (yarn.lock)
remote:        yarn install v0.28.1
remote:        [1/5] Resolving packages...
remote:        error An unexpected error occurred: "https://registry.yarnpkg.com/@luna%2fredacted: Not found".
remote:        info If you think this is a bug, please open a bug report with the information provided in "/tmp/build_2bb3cada459c6ac0aba00f8d4dce9259/yarn-error.log".
remote:        info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
remote: 
remote: -----> Build failed
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Deploy a project to heroku that uses private packages.

**What is the expected behavior?**
It should install private packages. If I revert to 0.27.5 it works fine.

**Please mention your node.js, yarn and operating system version.**
node v6, yarn 0.28.1

**Comment by Daniel15** [comment]
> When I deploy on heroku, it automatically uses the latest version 0.28.1

This sounds like a bug with Heroku. 0.28.1 is an RC release so I'd avoid using it in a production scenario for now.

> the deploy fails due to not being able to find private packages.

Is there an existing GitHub issue for this? If not, let's use this issue to track it.

**Comment by SpainTrain** [comment]
I believe this may have been introduced by https://github.com/yarnpkg/yarn/pull/3774/files

@AlastairTaft we fixed this issue in our CI & Heroku with the following change to `.npmrc`:

```
--- a/.npmrc
+++ b/.npmrc
@@ -1,2 +1,2 @@
 save-prefix='~'
-//registry.npmjs.org/:_authToken=${NPM_TOKEN}
+_authToken=${NPM_TOKEN}
```

> When I deploy on heroku, it automatically uses the latest version 0.28.1

FYI, you can set this in `engines` in `package.json` and Heroku will respect it - https://devcenter.heroku.com/articles/nodejs-support#build-behavior

**Comment by BYK** [comment]
This should be fixed on latest master. Anyone willing to confirm?

**Comment by SpainTrain** [comment]
@BYK 

Can confirm that private scoped modules install correctly on Heroku (nodejs buildpack) with the following `.npmrc` in project (yarn@1.0.1):

```ini
always-auth=true
https://registry.npmjs.org/:_authToken=${NPM_TOKEN}
```

**Comment by BYK** [comment]
@SpainTrain thanks! Still, it is not confirmed whether the fixes on latest master fix this or not.

I'm closing this for now since it most probably fixes by #4347.
